crates/cfxcore/core/src/light_protocol/error.rs

```diff
@@ -9,245 +9,218 @@ use crate::{
-error_chain! {
-    links {
-        Network(network::Error, network::ErrorKind);
-        StateDb(cfx_statedb::Error, cfx_statedb::ErrorKind);
-        Storage(cfx_storage::Error, cfx_storage::ErrorKind);
-    }
-
-    foreign_links {
-        Decoder(DecoderError);
-        Filter(FilterError);
-        AccountError(AccountError);
-    }
-
-    errors {
-        AlreadyThrottled(msg_name: &'static str) {
-            description("packet already throttled"),
-            display("packet already throttled: {:?}", msg_name),
-        }
-
-        ChainIdMismatch{ ours: ChainIdParamsOneChainInner, theirs: ChainIdParamsOneChainInner } {
-            description("ChainId mismatch"),
-            display("ChainId mismatch, ours={:?}, theirs={:?}.", ours, theirs),
-        }
-
-        ClonableErrorWrapper(error: ClonableError) {
-            description("Clonable error"),
-            display("{:?}", error.0.lock().to_string()),
-        }
-
-        GenesisMismatch{ ours: H256, theirs: H256 } {
-            description("Genesis mismatch"),
-            display("Genesis mismatch, ours={:?}, theirs={:?}.", ours, theirs),
-        }
-
-        InternalError(details: String) {
-            description("Internal error"),
-            display("Internal error: {:?}", details),
-        }
-
-        InvalidBloom{ epoch: u64, expected: H256, received: H256 } {
-            description("Logs bloom hash validation failed"),
-            display("Logs bloom hash validation for epoch {} failed, expected={:?}, received={:?}", epoch, expected, received),
-        }
-
-        InvalidHeader {
-            description("Header verification failed"),
-            display("Header verification failed"),
-        }
-
-        InvalidLedgerProofSize{ hash: H256, expected: u64, received: u64 } {
-            description("Invalid ledger proof size"),
-            display("Invalid ledger proof size for header {:?}: expected={}, received={}", hash, expected, received),
-        }
-
-        InvalidMessageFormat {
-            description("Invalid message format"),
-            display("Invalid message format"),
-        }
-
-        InvalidPreviousStateRoot{ current_epoch: u64, snapshot_epoch_count: u64, root: Option<StateRoot> } {
-            description("Invalid previous state root"),
-            display("Invalid previous state root for epoch {} with snapshot epoch count {}: {:?}", current_epoch, snapshot_epoch_count, root),
-        }
-
-        InvalidReceipts{ epoch: u64, expected: H256, received: H256 } {
-            description("Receipts root validation failed"),
-            display("Receipts root validation for epoch {} failed, expected={:?}, received={:?}", epoch, expected, received),
-        }
-
-        InvalidStateProof{ epoch: u64, key: Vec<u8>, value: Option<Vec<u8>>, reason: &'static str } {
-            description("Invalid state proof"),
-            display("Invalid state proof for key {:?} and value {:?} in epoch {}: {:?}", value, key, epoch, reason),
```

Comment:
Perhaps this should be: `display("Invalid state proof for key {:?} and value {:?} in epoch {}: {:?}", key, value, epoch, reason)`  swap `keys` and `values`

Reply:
Yes, it looks like a typo.